src/lib/ng-package/package.ts

```diff
@@ -67,10 +67,22 @@ export class NgPackage {
     return this.primary.$get('assets');
   }
 
+  /**
+   * @deprecated Use allowedNonPeerDependencies instead.
+   */
   public get whitelistedNonPeerDependencies(): string[] {
+    return this.primary.$get('whitelistedNonPeerDependencies') as string[];
+  }
+
+  public get allowedNonPeerDependencies(): string[] {
     // XX: default array values from JSON schema not recognized
     const defValue = ['tslib'];
-    const value = (this.primary.$get('whitelistedNonPeerDependencies') as string[]) || defValue;
+
```

Comment:
I don't think this logic is correct here, `whitelistedNonPeerDependencies` will always be truthy irrespective if it's being used or not.

I think something like the below should do
```
public get allowedNonPeerDependencies(): string[] {
  const allowedNonPeerDependencies = this.primary.$get('allowedNonPeerDependencies');
  if (this.primary.$get('allowedNonPeerDependencies').length) {
    return [...new Set(
      ...allowedNonPeerDependencies,
      'tslib',
    )];
  }

  return [...new Set(
    ...this.primary.$get('whitelistedNonPeerDependencies'),
    'tslib',
  )];
}
```

Reply:
Is it possible that the property coming from `$get()` could be `undefined`?  I like your changes on how the array is constructed, here's what I came up with:
```
  public get allowedNonPeerDependencies(): string[] {
    const alwaysInclude = ['tslib'];

    const allowedNonPeerDependencies = this.primary.$get('allowedNonPeerDependencies') as string[];
    
    // deprecated prop list
    const deprecated = this.whitelistedNonPeerDependencies;

    return Array.from(
      new Set([
        ...(deprecated ? deprecated : []),
        ...(allowedNonPeerDependencies ? allowedNonPeerDependencies :[]) ,
        ...alwaysInclude,
      ])
    );
  }
```
If it's true that this value will always be truthy then I can remove the trinary operators.

Reply:
The value cannot be undefined because a default is defined in the json schema.

Reply:
Ah ha!  Ok, thank you for helping me understand.  Appreciate your code review - learning a lot :)

Reply:
I think merging both the whitelist and the allowed list here isn’t correct.

Maybe use the new option when provided instead of the deprecated one. Basically invert the logic in my example above.

Reply:
I've made that change